symbolic-symcache/src/compat.rs

```diff
@@ -336,3 +327,19 @@ impl<'a> Iterator for Lines<'a> {
         }
     }
 }
+
+/// Helper to create neat snapshots for symbol tables.
+pub struct FunctionsDebug<'a>(pub &'a SymCache<'a>);
+
+impl fmt::Debug for FunctionsDebug<'_> {
```

Comment:
You might as well make this the `Debug` impl of `Functions` directly. Though that would exhaust the iterator, which is not nice. Not quite sure what the `Debug` expectations for iterators are. I think `Debug` impls are essentially free to do anything.

Reply:
I think I tried that; the problem is that you can't mutate things in a `Debug` impl.